src/index.ts

```diff
@@ -629,14 +629,13 @@ export default class Purchases {
    * @param {String?} userDefaultsSuiteName An optional string. iOS-only, will be ignored for Android. 
    * Set this if you would like the RevenueCat SDK to store its preferences in a different NSUserDefaults suite, otherwise it will use standardUserDefaults.
    * Default is null, which will make the SDK use standardUserDefaults.
-   * @returns {Promise<void>} Returns when setup completes
    */
   public static setup(
     apiKey: string,
     appUserID?: string | null,
     observerMode: boolean = false,
     userDefaultsSuiteName?: string
-  ) {
+  ): void {
```

Comment:
Just wondering about this one: currently the method does actually `return RNPurchases.setupPurchases(...)`, which in fact as per my testing is a `Promise<void>` (technically on Android it seems to return `Promise<null>`, whereas in Apples iOS case it is indeed just `Promise<void>`.

Reply:
🤔 
I think the best approach here would probably be to just remove the return and have this be a `void` function - there's really nothing to gain from a promise for this method, so the cleanest way to go about it would be to unify both to `void` and remove the `return`. What do you think?

Reply:
I think, the key question here would be, should the method return a Promise so that consumers become aware, if the `setup(...)` failed? I'm not familiar enough with the "guts" of the implementation to know, what such failure states would be, but for a consumer of the SDK, it would allow them to at least log an error to something like Instabug, if such a promise rejects, so there are no "silent fails".

Although, as we embarked on implementing the SDK for our App right now, it would also mean, that maybe the SDK documentation/guidance should reflect this as an async scenario.

Reply:
that's a great observation. From an SDK standpoint, this is a case where we really might want to directly crash if the initialization fails: since developers are supposed to initialize the SDK as early as possible within their app flow, the app flow might entirely be broken if the SDK isn't configured correctly, and there aren't many reasons for it to fail either, unless a big problem is encountered that should be fixed way before production. 

Do you think a promise + rejection would be preferable to an error in a case like that?

Reply:
Hard for me to answer, as I would answer it only from the specific implementation approach we intent to take. Can't really speak to all scenarios/current implementation practices.

For consideration, how should the setup step behave, if e.g. the public key is no longer valid?

In fairness, the current docs seem to imply that `setup(...)` is not an async method, i.e.:

```typescript
export default class App extends React.Component {
  
  componentDidMount() {
    Purchases.setDebugLogsEnabled(true);
    Purchases.setup("public_sdk_key");
  }
  
}
```

is the snippet in https://docs.revenuecat.com/docs/configuring-sdk

Reply:
yeah, the docs are correct, `setup` isn't async. if the public key isn't valid, currently, `setup` will produce warnings visible in the logs, however the SDK will continue to behave mostly as normal, except that the backend will reject requests

Reply:
So then it makes sense to leave it like that from a continuity perspective 😄

Reply:
@tomwanzek re-reading your comment, I'm not clear on what you'd recommend: leaving it async or sync?

Reply:
I'm leaving this without the Promise, since it's what our docs specify, the return type wasn't specified before, and it's not really an async method. Hope that makes sense!